src/test/java/harness/utils/CleanupTestUserWorkspaces.java

```diff
@@ -0,0 +1,115 @@
+package harness.utils;
+
+import bio.terra.cli.exception.UserActionableException;
+import bio.terra.cli.serialization.userfacing.UFWorkspace;
+import bio.terra.cli.serialization.userfacing.UFWorkspaceUser;
+import bio.terra.workspace.model.IamRole;
+import com.fasterxml.jackson.core.type.TypeReference;
+import harness.TestCommand;
+import harness.TestContext;
+import harness.TestUsers;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class implements a script to cleanup workspaces owned by test users. Tests are not supposed
+ * to leak workspaces, but if they do, the logging in this class should help track down any
+ * offenders. During development, in particular, workspace leaks are more likely, so this is a
+ * backup cleanup mechanism for those.
```

Comment:
This assumes that the test users are only used by the Terra CLI. If the test users create workspaces for other test use cases, those will also be deleted here.
Is that a safe assumption? Should we do some workspace tagging to check that it's a workspace created by the CLI tests?

Reply:
The `TestUsers` enum users are intended only for Terra CLI testing, so we should be able to delete everything they create.

My main concern is not that the test users would create workspaces for other use cases. It's that someone might share their workspace with a test user and it would get deleted. That's the reason for the `OWNER` check, but even that's only a partial safeguard because you could make a test user an `OWNER` on a workspace you manually create. I don't have a solution for this. It's the reason for running this manually and offering a dry-run option.

Tagging sounds great, but I'm not sure how to do that reliably. We can't use the name/description because the CLI tests need to test creating workspaces with neither, one, or both of those fields. Any other ideas/suggestions welcome!

Longer-term, it would be great if janitor could also clean up workspaces. I'm not sure exactly how that would work -- maybe WSM tells janitor about a new workspace if some flag in the request is set. The CLI could also tell the janitor, after a new workspace is created and we have the id.

Reply:
If these test users are created and noted to only be for Terra CLI testing, sgtm. If someone manually shares a workspace with a test user that's reserved for Terra CLI testing, that's on them, don't do that.

re: tagging. We'd need a WSM feature request to add metadata to workspaces. And the CLI 

For something to work with Janitor, we need it to be able to uniquely identify the resource before the resource is created. This could be done as a WSM client if WSM allows specifying a workspace UUID in the create request, or with an integration inside of WSM where it registers the to-be-created workspace UUID before starting creation based on a request flag or service config. If you can't identify the resource before it's created, you may get an error between creation and registration and then Janitor will never know something was supposed to be cleaned up.
Janitor also needs permissions to delete the resource. So in this case, we would also register the workspace uuid & the test user creating the workspace and assume we can give the Janitor a service account that can impersonate all test users.